spec/models/sufia/ability_spec.rb

```diff
@@ -38,16 +38,35 @@
     it { is_expected.to be_able_to(:read, ContentBlock) }
     it { is_expected.to be_able_to(:read, Sufia::Statistics) }
     it { is_expected.to be_able_to(:read, :admin_dashboard) }
-    it { is_expected.not_to be_able_to(:manage, AdminSet) }
+    it { is_expected.to be_able_to(:manage, AdminSet) }
     it { is_expected.to be_able_to(:create, AdminSet) }
   end
 
   describe "AdminSets and PermissionTemplates" do
     let(:permission_template) { build(:permission_template, admin_set_id: admin_set.id) }
     let(:permission_template_access) { build(:permission_template_access, permission_template: permission_template) }
-    describe "a user with edit access" do
-      let(:user) { create(:user) }
+    let(:user) { create(:user) }
+    let(:admin_set) { create(:admin_set) }
+
+    describe 'as admin' do
+      before { expect(ability).to receive(:admin?).and_return(true).at_least(1).times }
```

Comment:
This is probably causing your problem. The call to `ability` is invoking `Ability.new` (and all the methods that grant access) before the mock can be created.  Instead, try mocking on `RoleMap`